Update devops-command-center.rst (#4169)
mattermost/docs: source/administration/devops-command-center.rst

*Available in Mattermost Enterprise Edition E20, Mattermost Cloud Professional, and Mattermost Cloud Enterprise.*

> Incidents are situations which require an immediate response and benefit from a clearly defined process guiding toward resolution. Mattermost Incident Management equips your team to coordinate, manage, and resolve incidents from within Mattermost:

Reviewer: ```suggestion
Incidents are situations which require an immediate response, and incidents benefit from a clearly defined process that guides toward resolution. Mattermost Incident Management equips your team to coordinate, manage, and resolve incidents from within Mattermost:
```

* Starting an incident automatically creates a dedicated incident channel to coordinate discussion. 
  - Use custom categories in the left-hand sidebar to streamline access to ongoing incidents.
* Playbooks define the tasks toward resolution as your team manages the incident.
  - Define your playbooks in advance with automated actions such as creating a Jira ticket, starting a Zoom call, or finding out who is on-call in Opsgenie.
* Reviewing the timeline and conversation helps refine the process for future such incidents.
  - Export the incident channel conversation to conduct a post-mortem review.

> Delivering more reliable services requires an efficient resolution process. Use Mattermost Incident Management not only to resolve incidents more efficiently, but also to learn from past incidents and refine the process your team relies upon for success.

Reviewer: I really like this line. Not only does MM IM give you the tools to deliver more reliable services, but help you learn from the past!

.. contents::
  :depth: 1
  :local:
  :backlinks: entry
  
Installing Mattermost Incident Management
-----------------------------------------

*For Cloud deployments*

Mattermost Incident Management is included in the Mattermost Cloud workspace, enabled by default, and upgraded automatically.

*For self-managed deployments*

Mattermost Incident Management is available in the Plugin Marketplace:

> 1. As a system administrator, navigate to **Main Menu > Plugin Marketplace**.

Reviewer: ```suggestion
1. As a System Admin, navigate to **Main Menu > Plugin Marketplace**.
```

2. Search for **Incident Management**.
3. Select **Install** if not yet installed, and then **Configure** to enable.
4. From the plugin configuration page, set **Enable Plugin** to **true**.

> 5. Click **Save** to enable the plugin.

Reviewer: ```suggestion
5. Select **Save** to enable the plugin.
```

Getting Started with Incident Management
----------------------------------------

Mattermost Incident Management has the following components:

> * **Incidents** are situations that require an immediate response. Every reported incident results in a newly created, private incident channel exposing a custom user interface for collaboration and resolution.

Author: ```suggestion
* **Incidents** are situations that require an immediate response. Every reported incident results in a newly-created, public or private incident channel exposing a custom user interface for collaboration and resolution.
```

* **Slash commands** may be assigned to tasks defined in the playbook, helping to streamline common actions.

To access Mattermost Incident Management:

* From the Main Menu, select **Playbooks & Incidents** to configure playbooks and review incidents of which you are a member. System administrators have unrestricted access.
* From the channel header, click the **Incidents** icon to open the RHS and create a new incident or collaborate on active incidents of which you are a member.

Playbooks and Incidents
-----------------------

Playbooks and incidents are associated with teams in Mattermost. Starting an incident prompts for a name, description, and playbook. The name and description are used as the initial channel name and channel description, but may be modified thereafter. The selected playbook defines the stages and tasks, how incident statuses are communicated, and whether or not the incident channel is public or private. Read more about `public and private channels <https://docs.mattermost.com/help/getting-started/organizing-conversations.html>`_.

> Playbooks are always private, with membership in the playbook required to configure the playbook or start an incident using that playbook. Once an incident is started, however, any member of the team may be added to the incident channel. Users not in the team are unable to access playbooks or incidents belonging to that team.

Author: Excellent suggestion, thank you!

> Playbooks are always private, with membership in the playbook required to configure the playbook or start an incident using that playbook. Once an incident is started, however, any member of the team may be added to the incident channel. Users not in the team are unable to access playbooks or incidents belonging to that team.

Author: I've updated the text, @agarciamontoro.

> Playbooks are always private, with membership in the playbook required to configure the playbook or start an incident using that playbook. Once an incident is started, however, any member of the team may be added to the incident channel. Users not in the team are unable to access playbooks or incidents belonging to that team.

Author: ```suggestion
Playbooks are always private, and only playbook members can edit configurations or start an incident with the playbook. Once an incident is started any member of the team may be added to the incident channel. Playbooks and incidents are specific to a team and can't be accessed by members outside of that team.
```

Once an incident is ended, be sure to schedule a post-mortem meeting to iterate on the design of your playbook. Automatic posts can help expose a timeline of events for review, such as when a task is marked as complete or when the commander changes. You can export the messages from the incident channel and use them to build a post-mortem document. Identify bottlenecks by explaining gaps between task completion, and suggest changes to the playbook for the next incident.

Playbooks
^^^^^^^^^

.. contents::
  :backlinks: top
  :local:

Creating a playbook
~~~~~~~~~~~~~~~~~~~

A playbook must be defined before starting an incident.

1. Navigate to **Main Menu > Playbooks & Incidents**.
2. Start a **Blank Playbook**, or use the built-in **Incident Management Playbook** as a template.
3. Name your playbook and provide a description.

> 4. Within the **Tasks** tab, customize the stages and tasks:

Reviewer: Should there be a step here to select the **Tasks** tab ahead of the current step 4?

   * Create and name new tasks to capture actions your team should take to resolve the incident.
   * Create new stages to group tasks meant to be completed together.
   * Drag and drop to reorganize stages and tasks.
   * Optionally add task descriptions to give additional context to members of the incident channel. Descriptions support a limited form of markdown, including text styling and hyperlinks.
   * Optionally add a slash command to the task that can be executed by members of the incident channel at will.

> 5. Within the preferences tab, customize how status updates are communicated:

Author: ```suggestion
5. Within the **Preferences** tab, customize how status updates are communicated:
```

> 5. Within the **Preferences** tab, customize how status updates are communicated:

Reviewer: Should there be a step here to select the **Preferences* tab ahead of the current step 4?

   * Optionally configure a broadcast channel to which status updates will be copied. If you are not a member of the configured channel, `Unknown Channel` will be shown instead of the channel name.
   * Optionally configure the default reminder timer used to prompt for regular updates. The reminder timer may be changed when a status update is written.
   * Optionally configure a template to use for the first status update. Subsequent status updates will start with the text of the most recent update.

> 6. On the right hand side, configure the permissions:

Author: Right hand side of the screen in this instance. Which is in the same place as the RHS but can't be confused with the RHS.

> 6. On the right hand side, configure the permissions:

Author: So, perhaps "on the right-hand side of the screen"?

> 6. On the right-hand side of the screen, configure the permissions:

Reviewer: Nicely done! Very clear distinction from a RHS.

Working with stages and tasks
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Any member of the incident channel may work with stages and tasks:
* To mark a task as completed, select the unchecked checkbox next to the task. To undo this, clear the checkbox.
* To advance to the next stage, select **Next Stage**. Advancing to the next stage without completing the tasks in the stage prompts you to confirm.
* To assign a task to a member of the incident channel, select **No Assignee** (or the existing assignee's username), then select a user.
* To view any description associated with a task, select the information icon to the right of the task name.
* To execute a slash command associated with a task, select **Run** next to the listed slash command. Configured slash commands may be run as often as necessary.

Changing commanders
~~~~~~~~~~~~~~~~~~~

To change commanders, use any one of the following:

* Run the ``/incident commander @username`` slash command from the incident channel.
* From the desktop or browser:
  1. Select the **Incidents** icon in the channel header to open the right-hand sidebar.

>   2. Browse to the **Summary** tab.

Reviewer: Does the user browse to a tab or select a tab? The answer to this will impact additional sections of this page.

>   2. Browse to the **Summary** tab.

Author: ```suggestion
  2. Select the **Summary** tab.
```

  3. Within the **Summary** tab, select the current commander's name in the right-hand sidebar, then select the new commander.
    - To change the commander to a user who is not in the channel, first add the user to the channel.

Providing a status update
~~~~~~~~~~~~~~~~~~~~~~~~~

Incident status updates ensure that stakeholders remain informed about the progress toward incident resolution. To post a status update:
* From the desktop or browser:
  1. Select the **Incidents** icon in the channel header to open the right-hand sidebar.

>   2. Browse to the **Summary** tab.

Reviewer: Browse vs. select

>   2. Browse to the **Summary** tab.

Author: ```suggestion
  2. Select the **Summary** tab.
```

>   3. Within the **Summary** tab select **Update Status**.

Reviewer: ```suggestion
  3. Within the **Summary** tab, select **Update Status**.
```

    - If this is the first status update and the playbook defined a template, that template will be pre-populated here.
    - If this is a subsequent status update, the message from the last status update will be pre-populated here.
  5. Optionally set a reminder to prompt for the next status update.
    - If this is the first status update and the playbook defined a default reminder timer, that timer will be pre-selected here.
    - If this is a subsequent status update, the last reminder timer will be pre-populated here.
  6. Click **Update Status** to post your status update.
    - Status updates are posted to the incident channel as a message from the user providing the status update.

>     - If the playbook defined a broadcast channel, status updates are copied to the broadcast channel as a message from the incident bot.

Author: It's incident BOT in the UI. 
<img width="403" alt="Screen Shot 2020-12-17 at 18 25 46" src="https://user-images.githubusercontent.com/4223530/102514561-4bbe4e80-4095-11eb-86a8-330b001767b4.png">

Other bots have the same format.
<img width="383" alt="Screen Shot 2020-12-17 at 18 23 21" src="https://user-images.githubusercontent.com/4223530/102514359-0ac63a00-4095-11eb-9b76-265d140fbbc3.png">

I'm inclined to retain "incident bot". Thoughts?

>     - If the playbook defined a broadcast channel, status updates are copied to the broadcast channel as a message from the incident bot.

Author: The images got switched around, apologies for that.

>     - If the playbook defined a broadcast channel, status updates are copied to the broadcast channel as a message from the incident bot.

Author: Oh, in the popup the casing is different:
<img width="253" alt="Screen Shot 2020-12-17 at 18 37 14" src="https://user-images.githubusercontent.com/4223530/102515832-e4a19980-4096-11eb-8f12-cdfc6080b125.png">

    - If the playbook has a defined broadcast channel, status updates are copied to the broadcast channel as a message from the incident bot.

The most recent status post will also appear in the right-hand sidebar of the incident channel. To correct or remove a status post, edit or delete the post as needed. Status updates that are broadcast to another channel will not be edited or deleted.

Ending an incident
~~~~~~~~~~~~~~~~~~~

Incident members can end an incident using any one of the following:

> * Run the ``/incident end`` slash command from within the incident channel.

Reviewer: ```suggestion
* Within the incident channel, run the ``/incident end`` slash command.
```

* From the desktop or browser:
  1. Select the **Incidents** icon in the channel header to open the right-hand sidebar.

>   2. Browse to the **Tasks** tab.

Reviewer: Browse vs. select

>   2. Browse to the **Tasks** tab.

Author: ```suggestion
  2. Select the **Tasks** tab.
```

  3. Within the **Tasks** tab, if you are on the final stage, select **End Incident**.
  4. Otherwise, to end the incident from an earlier stage, select the **...** menu, then select **End Incident**.

Ending an incident signals to all members of the channel that the incident is no longer ongoing. Members of the team can continue to post in the channel, mark tasks as complete, and change the commander if needed.

Restarting an incident
~~~~~~~~~~~~~~~~~~~~~~

If an incident was ended prematurely, it can be restarted using any one of the following:

> * Run the ``/incident restart`` slash command from within the incident channel.

Reviewer: ```suggestion
* Within the incident channel, run the ``/incident restart`` slash command.
```

* From the desktop or browser:
  1. Select the **Incidents** icon in the channel header to open the RHS.

>   2. Browse to the **Tasks** tab.

Reviewer: Browse vs. select

>   2. Browse to the **Tasks** tab.

Author: ```suggestion
  2. Select the **Tasks** tab.
```

Ending an incident signals to all members of the channel that the incident is no longer ongoing, but conversation in the channel, marking tasks as complete, and even changing the commander are all still allowed.

Restarting an incident
^^^^^^^^^^^^^^^^^^^^^^
~~~~~~~~~~~~~~~~~~~~~~

An ended incident can be restarted at any time using ``/incident restart`` from within the incident channel or via **Restart Incident** in the RHS. Some playbooks may define stages and tasks to complete after an incident has been resolved, such as scheduling and completing a post-mortem.
If an incident was ended prematurely, it can be restarted using any one of the following:
* Run the ``/incident restart`` slash command from within the incident channel.
* From the desktop or browser:
  1. Select the **Incidents** icon in the channel header to open the RHS.
  2. Browse to the **Tasks** tab.
  3. Click **Restart Incident**.

Incident status and information
-------------------------------
Reviewing past incidents
~~~~~~~~~~~~~~~~~~~~~~~~

> To view the status of your active incident(s) select **Main Menu > Playbooks & Incidents**. Select the **Incident** tab to view a list of incidents in your team. Select the incident name to view a summary of the incident, jump to the channel, or export the channel.

Author: Agree and noted.

> 3. Within the **Incidents** tab find the incident to be reviewed, then select the name of the incident.

Reviewer: ```suggestion
3. Within the **Incidents** tab, find the incident to be reviewed, then select the name of the incident.
```

> The ``/incident`` slash command allows interacting with incidents via the post textbox on desktop, browser, and mobile:

Reviewer: ```suggestion
The ``/incident`` slash command allows interaction with incidents via the post textbox on desktop, browser, and mobile for:
```

> To quickly Mattermost Incident Management, use the following test commands to create incidents populated with random data:

Reviewer: ```suggestion
To test Mattermost Incident Management, use the following test commands to create incidents populated with random data:
```

> - ``/incident test create-incident [playbook ID] [timestamp] [incident name]`` - Provide the ID of an existing playbook to which the current user has access, a timestamp, and an incident name. The command creates an ongoing incident with the creation date set to the specified timestamp.

Reviewer: If there's a specific timestamp format required, consider specifying that format.

Author: @agarciamontoro, is there a specific format required?

> - ``/incident test bulk-data [ongoing] [ended] [start date] [end date] [seed]`` - Provide a number of ongoing and ended incidents, a start and end date, and an optional random seed. The command creates the given number of ongoing and ended incidents, with creation dates randomly between the start and end dates. The seed may be used to reproduce the same outcome on multiple invocations. Incident names are generated randomly. 

Reviewer: ```suggestion
- ``/incident test bulk-data [ongoing] [ended] [start date] [end date] [seed]`` - Provide a number of ongoing and ended incidents, a start and end date, and an optional random seed. The command creates the specified number of ongoing and ended incidents, with creation dates set randomly between the specified start and end dates. The seed may be used to reproduce the same outcome on multiple invocations. Incident names are generated randomly. 
```

Reviewer: Is there value in specifying where seed values come from, where users can find/use them?

Author: Good question. I'll defer to @agarciamontoro on that.

Reviewer 2: Seed values are simply a user-specified number, they don't have to follow any pattern whatsoever. The gist of it is that I will get the exact same output if and only if I provide the exact same seed value (whichever it is).

Reviewer: Thanks, @agarciamontoro, for this context! Can a seed value be any combination of letters and numbers? Can it be any character length? If so, great! If there are any qualifiers around what a seed value can contain, it may be worth mentioning.

Glossary
--------

- **Commander:** The Mattermost user currently responsible for transitioning an incident from ongoing to ended.
- **Incident:** An event requiring the coordinated actions of one or more Mattermost users. An incident is either ongoing or ended.
- **Incident channel:** A Mattermost channel dedicated to real-time conversation about the incident.
- **Incident insight page:** The incident details and analytics page, which also provides the channel export download link. It is not available on mobile.
- **Incident member:** A Mattermost user with access to the corresponding incident channel.
- **Playbook:** A task-based process that's followed in order to resolve an incident.
- **Playbook configuration page:** The playbook configuration and editing page. It is not available on mobile.
- **Stage:** A set of tasks grouped together to achieve a specific goal of the workflow, which generally need to be completed before proceeding to the next stage of the incident resolution process.
- **Tasks:** The individual steps required to complete the stages of an incident. Tasks can optionally be assigned to specific incident participants into stages.

> - **The RHS:** The incident list and incident details displayed on the right hand side (RHS) of the web app. It is not available on mobile.

Reviewer: If this page will contain a Glossary that explains the RHS, then please feel free to ignore the suggestions where I've written out the acronym in full.

Author: The glossary has been removed so I've committed your suggestions. :)

> *For self-managed deployments*

Author: @lieut-data, I know IM is pre-packaged so just wanted to confirm this is still accurate for self-managed deployments.

Mattermost Incident Management is included in the Mattermost Cloud workspace and is enabled by default.

Getting Started with Incident Management
----------------------------------------

Mattermost Incident Management has the following components:

* **Slash commands:** Used to perform actions and manage incidents.
* **Playbooks:** A playbook is a recipe that defines how an incident is started and resolved, as well as the retrospective process. Playbooks also include configuration options such as reminders.
- **Stages:** A set of tasks grouped together to achieve a specific goal of the workflow, which generally need to be completed before proceeding to the next stage of the incident resolution process.

> - **Tasks:** The individual steps required to complete the stages of an incident. Tasks can optionally be assigned to specific incident participants into stages.

Reviewer 3: ```suggestion
 - **Tasks:** The individual steps required to complete the stages of an incident. Tasks can optionally be assigned to specific incident participants. 
```

During an active incident, you want to focus on triaging and solving the problem as soon as possible. Planning your incident support strategy ahead of time with playbooks is the best way to ensure incidents run smoothly. A playbook is a recipe for dealing with and resolving an incident. In a playbook, you can plan ahead so that during an incident responders know exactly what to do.

Within each playbook, you can create stages and tasks to ensure that items are addressed and completed in sequential order. The tasks can optionally be associated with slash commands and assigned to individual team members. Playbooks also include status update settings, including a reminder feature, so regular updates can be shared to selected channels at set intervals. Make sure to schedule a retrospective analysis to iterate on the design of your playbooks after the incident finishes.

Once complete, incident channels can be exported using the channel export option for analysis. Teams can identify bottlenecks in the incident by seeing time gaps between when checklist items are completed and incorporating necessary changes into the playbook for next incident.

> Incident status updates

Author: @lieut-data, I've added this here because it doesn't fall squarely under just playbooks or just incidents. I'm 0/5 on where it lives though, so feedback welcome.

7. Share this playbook with other members of the team to allow them to use the playbook to start an incident, as well as edit the contents.

Editing a playbook
~~~~~~~~~~~~~~~~~~

You can edit a playbook at any time. However, the changes will only be applied to future incidents - not the active incidents, or incidents that previously used that playbook.

Navigate to **Main Menu > Playbooks & Incidents** and select the playbook you'd like to edit.

**Permissions:**

- Change the channel type created with this playbook.
- Share the playbook.

**Stages and tasks:**

- Delete a stage and its associated tasks.
- Add new tasks to an existing stage.
- Edit tasks in an existing stage.
- Edit the slash commands in existing tasks.
- Add new stages and tasks.

**Preferences:**

> - Select a broadcast channel where status updates will be automatically posted.

Reviewer 3: ```suggestion
 - Select an optional broadcast channel where incident status updates will be automatically copied. Incident status updates always post to the incident channel. 
```

> - Set a timer to reminder the incident commander to provide a status update.

Reviewer 3: ```suggestion
 - Set a default timer interval reminding the incident commander to provide a status update. 
```

> - Add a message template with the reminder text.

Reviewer 3: ```suggestion
 - Add a message template used for the first status update.
```

Deleting a playbook
~~~~~~~~~~~~~~~~~~~

> You can delete a playbook provided it's not currently being used in an active incident.

Reviewer 3: This should not actually be true: you can delete a playbook at any time.
